### DISCUSSION TITLE
Completions Pwsh

### DISCUSSION THREAD

**Discussion by pbl987** [solokeys/solo2-cli#80] [Q&A] [answered]
What do i do with the file 'solo2.completions.powershell' ?

**Comment by macxcool** [comment, marked as answer]
Completions would, normally, go in your profile.ps1 file to be loaded when you start Powershell. You can use the $profile variable to find out where your profile script should be. Where does the solo2.completions.powershell file come from?

It probably works the same as other completions. There's a discussion here about the github completions that might help.

**Reply by pbl987** [reply]
it is found here in assets:
https://github.com/solokeys/solo2-cli/releases/tag/v0.2.0
Where is the discussion you are talking about?

The contents are too big to copy and paste them. I tried using `type C:\Users\patrick\Documents\PowerShell\solo2.completions.powershell | iex` and other stuff without luck. I can't run scripts without ps1 extension, so i guess i am doing something wrong.

**Reply by macxcool** [reply]
I'm sorry. I didn't include a link. Here: https://github.com/cli/cli/issues/695
I'll have a look and see what works well, but on the surface of it, I think adding something like this to your profile might work.
```
Invoke-Command -ScriptBlock {
    . C:\Users\patrick\Documents\PowerShell\solo2.completions.powershell
}
```
That's dot-sourcing the file, so it's dot-space...

**Reply by pbl987** [reply]
It works, but first i have to rename the file to `solo2.completions.powershell**.ps1**` 
If i don't, it opens the "Open with"-dialogue of windows, where i have to choose an application.

> so it's dot-space...

What do you mean?

**Comment by macxcool** [comment]
dot-space, means that in order to load a powershell script without running it, you have to type a dot (i.e. '.') and a space, then the path to the script.

For example:
. c:\Temp\solo2.completions.powershell.ps1
 /\ note the space between the '.' and the path.

And yes, powershell scripts normally have a '.ps1' extension, so it makes sense that you have to name it that.
-- Mark C
 Sent with Tutanota, enjoy secure & ad-free emails. 



Aug 2, 2022, 4:14 p.m. by ***@***.***:

**Reply by pbl987** [reply]
I just wonder why the file isn't named properly in the beginning. It seems to me like we are not using it as intended. 
Otherwise i would mark your post as answer.

**Comment by macxcool** [comment]
You may be right, but since nobody who works on the project is answering I'm just guessing based on my knowledge of PowerShell and how it works.

The question is, did it work? ;-)
-- Mark C
 Sent with Tutanota, enjoy secure & ad-free emails. 



Aug 2, 2022, 4:43 p.m. by ***@***.***:

**Reply by pbl987** [reply]
I am just an end user, and when i pressed TAB, it did autocomplete. 
So my guess is, it does work ;-)
